src/ethereum_test_types/blob.py

```diff
@@ -0,0 +1,469 @@
+"""Blob class and functionality."""
+
+import random
+from enum import Enum
+from hashlib import sha256
+from os.path import realpath
+from pathlib import Path
+from typing import List
+
+import ckzg
+
+from ethereum_test_base_types.base_types import Bytes, Hash
+from ethereum_test_base_types.pydantic import CamelModel
+from ethereum_test_forks import (
+    # BLOB_BASE_FEE_UPDATE_FRACTION_PRAGUE,
+    BLS_MODULUS,
+    BYTES_PER_BLOB,
+    BYTES_PER_COMMITMENT,
+    BYTES_PER_FIELD_ELEMENT,
+    # BYTES_PER_PROOF,
+    # CELL_LENGTH,
+    FIELD_ELEMENTS_PER_BLOB,
+    KZG_ENDIANNESS,
+    # MAX_BLOB_GAS_PER_BLOCK,
+    # MAX_BLOBS_PER_BLOCK_ELECTRA,
+    # TARGET_BLOB_GAS_PER_BLOCK,
+    # TARGET_BLOBS_PER_BLOCK_ELECTRA,
+    Cancun,
+    Fork,
+    Osaka,
+    Prague,
+)
+
+TRUSTED_SETUP_FILE_NAME = "blob_trusted_setup.txt"
+TRUSTED_SETUP_PATH = Path(realpath(__file__)).parent / TRUSTED_SETUP_FILE_NAME
+TRUSTED_SETUP = ckzg.load_trusted_setup(str(TRUSTED_SETUP_PATH), 0)
+# print(f"{TRUSTED_SETUP_FILE_NAME} has been loaded")
+
+
+def get_eest_root_folder(marker_files=("pyproject.toml", ".git", "tests", "src")) -> Path:
+    """Search for a folder where all files/folders listed above exist (root of project)."""
+    current = Path(__file__).resolve()
+    for parent in current.parents:
+        if all((parent / marker).exists() for marker in marker_files):
+            return parent
+    raise RuntimeError("Project root folder of execution-spec-tests was not found")
+
+
+eest_root = get_eest_root_folder()
+
+
+class Blob(CamelModel):
+    """Class representing a full blob."""
+
+    data: Bytes
+    commitment: Bytes
+    proof: List[Bytes] | Bytes  # Bytes < Osaka, List[Bytes] >= Osaka
+    cells: List[Bytes] | None  # None (in json: null)  < Osaka, List[Bytes] >= Osaka
+
+    versioned_hash: Hash
+    name: str  # blob_<fork>_<seed>
+    fork: str
+    seed: int
+    timestamp: int
+
+    @staticmethod
+    def get_filepath(fork: str, seed: int, timestamp: int):
```

Comment:
I think we can improve this even further by, instead of passing the fork name, we instead pass the characteristics that make the blob unique, e.g.
```suggestion
    def get_filepath(kzg_cell_proofs: int, seed: int):
```
Now Cancun and Prague blobs will resolve to the same file name because they both use `kzg_cell_proofs = 0`, and Osaka and future forks use `kzg_cell_proofs = 128`.

Reply:
What problem does this solve? I intentionally named the blobs so that from the filename alone I can tell which fork we created them for. Yes Cancun and Prague use the same blob format/rules but it is not harmful to have the info which blob they were created for in the filename. I feel like a common theme of our repository is having the same name re-used in different contexts (e.g. types.py in multiple locations with different contents) which makes everything harder to understand.

I do agree that the resulting .json files should not be pushed to the repository, so the local duplication (e.g. there might be a case where you generate the same blob for Cancun and Prague and the only difference in their fork name) is not an issue IMO.

Reply:
It solves duplication of work because timestamp and fork indirectly affect the output (timestamp only in the case of transition forks), but we can reduce the amount of produced files by using the variables that directly affect the output, namely the cell proof count.

So for example if you have two tests and both reference the blobs with seeds 1 and 2:
```python
@pytest.mark.valid_at_transition_to("Osaka")
def test_1(fork: Fork):
    blob_1 = Blob(fork=fork, seed=1, timestamp=14_999)
    blob_0 = Blob(fork=fork, seed=2, timestamp=15_000)

@pytest.mark.valid_from("Prague")
def test_2(fork: Fork):
    blob_1 = Blob(fork=fork, seed=1, timestamp=1000)
    blob_0 = Blob(fork=fork, seed=2, timestamp=2000)
```
With the fork-name/timestamp/seed approach you will generate files:
```
blob_PragueToOsakaAtTime15k_1_14999.json
blob_PragueToOsakaAtTime15k_2_15000.json
blob_Prague_1_1000.json
blob_Prague_2_2000.json
blob_Osaka_1_1000.json
blob_Osaka_2_2000.json
```
And the cell-proofs/seed approach you will generate:
```
blob_1_cell_proofs_0.json
blob_2_cell_proofs_0.json
blob_1_cell_proofs_128.json
blob_2_cell_proofs_128.json
```
So it's going to grow with the amount of forks and the amount of tests (due to the timestamp).